### PR TITLE
Drupal 8 port

### PR DESCRIPTION
**Pull request by vijaycs85**

### CODE REVIEW

File: js/form.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    attach: function attach() {
+      if (typeof $.fn.drupalSetSummary === 'undefined') {
+        return;
+      }
+
+      // Channels and sections checkboxes.
+      $('input[data-channel-id]').each(function() {
+        var _channel_id = $(this).data('channel-id');
+        if (!$(this).attr('checked')) {
+          // $('input[data-section-of="' + _channel_id + '"]').parent().hide();
```
Comment (juampynr) [inline js/form.js]: Was this copied from another place? If not, shall we remove it if it is not needed?

File: src/Publisher.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  public function getChannel($channel_id) {
+    $response = $this->publisher()->get('/channels/{channel_id}', ['channel_id' => $channel_id]);
+    return $this->handleResponse($response);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getArticle() {
+
```
Comment (juampynr) [inline src/Publisher.php]: Any reason why this and the following are empty? Perhaps adding a `@TODO` might help?

**Comment by juampynr** [comment]
Static review looks fine to me. I just posted a couple questions.

@jazzdrive3 @m4olivei ? Thoughts on this pull request? Should this be ported to https://www.drupal.org/project/applenews Drupal.org? That would make the tests to run automatically.

File: src/Publisher.php (hunk 2 above)
Comment (vijaycs85) [reply src/Publisher.php]: @juampynr my bad. We are going to fill all these methods before the merge.

File: js/form.js (hunk 1 above)
Comment (vijaycs85) [reply js/form.js]: yeah, I took from D7.  I will clean this one up.

**Comment by m4olivei** [comment]
@juampynr I don't have time to go through this now unfortunately. I trust you all are killing it. Yes to porting to Drupal.org. How do you want to approach that? I have maintainer rights, so I could mirror this work to d.o manually, but I don't really have time to stay on top of it right now, and I don't want to be a blocker. I also don't have access to grant maintainership, but the two Chapter 3 gentlemen who do are very approachable and would probably be open to handing out maintainership to @vijaycs85. If you need a hand reaching out to them, let me know.

**Comment by vijaycs85** [comment]
> would probably be open to handing out maintainership to @vijaycs85

@m4olivei could we have git level maintainer access? I am trying to use the port patch in https://www.drupal.org/project/applenews/issues/2846081 but it's failing as 8.x-1.x doesn't have a release yet.

**Comment by juampynr** [comment]
In the meantime, I am merging this.

**Comment by m4olivei** [comment]
Added the 8.x-1.x-dev release to the applenews project on d.o.

**Pull request merged**